File: app-backend/models/Invoice.js
const mongoose = require('mongoose')

const invoiceSchema = new mongoose.Schema({
  rate: {
    type: Number,
    required: true
  },
  quantity: {
    type: Number,
    required: true
  },
  products: {
    type: String,
    required: true
  },
  date: {
    type: Date,
    default: Date.now()
  },
  challan: {
    type: String,
    default: null
  }
})

module.exports = mongoose.model('Invoice', invoiceSchema)

File: app/lib/screens/InvoicePage/models/Invoice.dart
class Invoice {
  String id;
  String product;
  double rate;
  double quantity;
  String challan;
  String date;

  Invoice(
      this.id, this.product, this.rate, this.quantity, this.challan, this.date);

  Invoice.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    product = json['products'];
    rate = json['rate'].toDouble();
    quantity = json['quantity'].toDouble();
    challan = json['challan'];
    date = json['date'].toString();
  }
}
